pkg/kube/pod_runner_test.go

```diff
@@ -109,9 +114,84 @@ func (s *PodRunnerTestSuite) TestPodRunnerForSuccessCase(c *C) {
 	cancel()
 }
 
+// TestPodRunnerWithDebugLabelForSuccessCase adds a debug entry into the context and verifies the
+// pod got created with corresponding label using the entry or not.
+func (s *PodRunnerTestSuite) TestPodRunnerWithDebugLabelForSuccessCase(c *C) {
+	jobIDSuffix := "JobID"
+	for _, tc := range []struct {
+		name          string
+		targetKey     string
+		contextKey    string
+		contextValue  string
+		expectedLabel bool
+	}{
+		{
+			name:          "target key (kanister.io/JobID) present in pod labels",
+			targetKey:     path.Join(consts.LabelPrefix, jobIDSuffix),
+			contextKey:    path.Join(consts.LabelPrefix, jobIDSuffix),
+			contextValue:  "xyz123",
+			expectedLabel: true,
+		},
+		{
+			name:          "target key (kanister.io/JobID) not present in pod labels",
+			targetKey:     path.Join(consts.LabelPrefix, jobIDSuffix),
+			contextKey:    path.Join(consts.LabelPrefix, "NonJobID"),
+			contextValue:  "some-other-value",
+			expectedLabel: false,
+		},
+	} {
+		ctx, cancel := context.WithCancel(context.Background())
+		ctx = field.Context(ctx, tc.contextKey, tc.contextValue)
+		cli := fake.NewSimpleClientset()
+		cli.PrependReactor("create", "pods", func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+			return false, nil, nil
+		})
+		cli.PrependReactor("get", "pods", func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+			p := &corev1.Pod{
+				Status: corev1.PodStatus{
+					Phase: corev1.PodRunning,
+				},
+			}
+			return true, p, nil
+		})
+		po := &PodOptions{
+			Namespace: podRunnerNS,
+			Name:      podName,
+			Command:   []string{"sh", "-c", "tail -f /dev/null"},
+		}
+		deleted := make(chan struct{})
+		cli.PrependReactor("delete", "pods", func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+			c.Log("Pod deleted due to Context Cancelled")
+			close(deleted)
+			return true, nil, nil
+		})
+		AddLabelsToPodOptionsFromContext(ctx, po, tc.targetKey)
+		pr := NewPodRunner(cli, po)
+		errorCh := make(chan error)
+		go func() {
+			_, err := pr.Run(ctx, afterPodRunTestKeyPresentFunc(tc.targetKey, tc.expectedLabel, deleted))
+			errorCh <- err
+		}()
+		deleted <- struct{}{}
+		c.Assert(<-errorCh, IsNil)
+		cancel()
+	}
+}
+
 func makePodRunnerTestFunc(ch chan struct{}) func(ctx context.Context, pc PodController) (map[string]interface{}, error) {
 	return func(ctx context.Context, pc PodController) (map[string]interface{}, error) {
 		<-ch
 		return nil, nil
 	}
 }
+
+func afterPodRunTestKeyPresentFunc(labelKey string, expectedLabel bool, ch chan struct{}) func(ctx context.Context, pc PodController) (map[string]interface{}, error) {
+	return func(ctx context.Context, pc PodController) (map[string]interface{}, error) {
+		<-ch
+		_, got := pc.Pod().Labels[labelKey]
+		if got != expectedLabel {
+			return nil, errors.New("Got different label than expected")
+		}
+		return nil, nil
+	}
+}
```

Comment:
We can also add an additional validation to check the label value
```suggestion
func afterPodRunTestKeyPresentFunc(labelKey string, isLabelExpected bool, expectedLabelValue string, ch chan struct{}) func(ctx context.Context, pc PodController) (map[string]interface{}, error) {
	return func(ctx context.Context, pc PodController) (map[string]interface{}, error) {
		<-ch
		labelValue, found := pc.Pod().Labels[labelKey]
		if found != isLabelExpected {
			return nil, errors.New("Found unexpected label")
		}
		if labelValue != expectedLabelValue {
		        return nil, errors.New("Found label doesn't match with expected label")
		}
		return nil, nil
	}
}
```

Reply:
Done